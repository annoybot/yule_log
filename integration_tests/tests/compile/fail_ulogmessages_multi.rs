use yule_log::{ULogMessages, ULogData};

#[derive(ULogData)]
pub struct A { value: u64 }
#[derive(ULogData)]
pub struct B { value: u64 }

#[derive(ULogMessages)]
pub enum LoggedMessages {
    #[yule_log(forward_other)]
    A(A),
    #[yule_log(forward_other)]
    B(B),
}

fn main() {}
