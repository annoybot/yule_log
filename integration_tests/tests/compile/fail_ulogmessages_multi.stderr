error: Only one variant may have #[yule_log(forward_other)].
  --> tests/compile/fail_ulogmessages_multi.rs:12:5
   |
12 | /     #[yule_log(forward_other)]
13 | |     B(B),
   | |________^
